src/components/Leaderboard.jsx

```diff
@@ -22,13 +24,18 @@ function Leaderboard() {
   }
 
   const handleNameSubmit = (name) => {
-    addScore(name, score);
-    setIsDisplayingForm(false);
+    if (score === 0) {
+      setIsDisplayingForm(false);
+    } else {
+      addScore(name, score);
+      getData();
+      setIsDisplayingForm(false);
+    }
   };
 
   return (
     <div className="container-gameOver">
-      {isDisplayingForm && <Form onSubmit={handleNameSubmit} />}
+      {score !== 0 && isDisplayingForm && <Form onSubmit={handleNameSubmit} />}
```

Comment:
can you explain the changes on this line before I approve this code? otherwise it's good to go!